worker/worker.go

```diff
@@ -526,36 +526,39 @@ func (w *worker) rhpScanHandler(jc jape.Context) {
 		defer cancel()
 	}
 
+	// defer scan result
+	var err error
 	var settings rhpv2.HostSettings
+	var priceTable rhpv3.HostPriceTable
+	defer func() {
+		w.recordScan(rsr.HostKey, priceTable, settings, err)
+	}()
+
+	// fetch the host settings
 	start := time.Now()
-	pingErr := w.withTransportV2(ctx, rsr.HostKey, rsr.HostIP, func(t *rhpv2.Transport) (err error) {
-		settings, err = RPCSettings(ctx, t)
+	err = w.withTransportV2(ctx, rsr.HostKey, rsr.HostIP, func(t *rhpv2.Transport) (err error) {
+		settings, err = RPCSettings(ctx, t, rsr.HostIP)
```

Comment:
Maybe slightly nicer to override the return value instead of passing the address in alongside the transport.

Reply:
Yeah passing it in was unfortunate but had no other option. I was considering adding it to the transport but I don't really think Luke would go for it since it doesn't really make sense to expose it on the transport either. I don't _really_ see how it's nicer to override the return value since that means I have to duplicate the override code but i'll make the change 😁

Reply:
Maybe it makes sense to add `RemoteAddress` though, on the `Transport`, returning the remote address of the underlying connection. It makes sense to expose that on the transport and we could use it as value to overwrite the `NetAddress` with inside `RPCSettings`...